components/region_cache_memory_engine/src/gc.rs

```diff
@@ -0,0 +1,524 @@
+// Copyright 2023 TiKV Project Authors. Licensed under Apache-2.0.
+
+use core::slice::SlicePattern;
+use std::{fmt::Display, sync::Arc};
+
+use engine_traits::{CacheRange, CF_DEFAULT, CF_WRITE};
+use skiplist_rs::Skiplist;
+use slog_global::{info, warn};
+use txn_types::{Key, WriteRef, WriteType};
+
+use crate::{
+    engine::EVICTION_KEY_BUFFER_LIMIT,
+    keys::{decode_key, encoding_for_filter, InternalKey, InternalKeyComparator},
+    memory_limiter::GlobalMemoryLimiter,
+    RangeCacheMemoryEngine,
+};
+
+// Split a ts encoded key, return the user key and timestamp.
+fn split_ts(key: &[u8]) -> Result<(&[u8], u64), String> {
```

Comment:
```suggestion
///  Try to extract the key and `u64` timestamp from `encoded_key`. 
///
/// See also: [`txn_types::Key::split_on_ts_for`]
fn split_ts(encoded_key: &[u8]) -> Result<(&[u8], u64), String> {
```

General idea: since this function now appears in two places (compaction_filter.rs and here), might make sense to just place it in `txn_types` to remove duplication. Leaving the naming of the function to you.